lua/autorun/client/cl_chip_lister.lua

```diff
@@ -0,0 +1,321 @@
+include( "cfc_chip_lister/shared/sh_chip_lister.lua" )
+
+local ID_WORLD = "[WORLD]"
+local CPUS_FORMAT = "%05d"
+local STR_GLOBAL = "Overall Total CPUs: "
+local STR_TOTAL = "Total: "
+local STR_CPUS = " CPUs: "
+local STR_MICROSECONDS = utf8.char( 181 ) .. "s"
+local STR_TITLE = "-----E2/SF Lister-----"
+local STR_TOGGLE = "(Press " .. string.upper( input.LookupBinding( "+use" ) or "e" ) .. "/use to toggle)"
+local STR_ENABLE = "Press " .. string.upper( input.LookupBinding( "+use" ) or "e" ) .. "/use to turn on"
+local STR_WAITING = "Waiting for next update from the server..."
+local RENDER_TARGET_NAME = "cfc_chiplister_rt"
+local MATERIAL_NAME = "cfc_chiplister_screen"
+local FONT_NAME = "CFC_ChipLister_Font"
+local FONT_SIZE = 30
+local MAX_ELEMENTS = 30
+local SCREEN_SIZE = 1024
+local SCREEN_SIZE_HALF = SCREEN_SIZE / 2
+local TOGGLE_DIST = 1500
+local COLOR_BACKGROUND = Color( 0, 0, 0, 255 )
+local COLOR_DIVIDER = Color( 255, 255, 255, 255 )
+local COLOR_TEXT = Color( 255, 255, 255, 255 )
+local COLOR_TEXT_FADED
+local COLOR_MICROS
+local COLOR_WORLD = Color( 150, 120, 120, 255 )
+local COLOR_FADE_OFFSET = Color( 0, 0, -0.65 )
+local COLOR_FADE_MICROS_OFFSET = Color( 0, 0, -0.25 )
+local CHIP_COLORS = {
+    E2 = Color( 216, 34, 45, 255 ),
+    SF = Color( 55, 100, 252, 255 ),
+}
+
+local rtChipLister = GetRenderTarget( RENDER_TARGET_NAME, SCREEN_SIZE, SCREEN_SIZE )
+local INFO_OFFSET_OWNER = 0
+local INFO_OFFSET_CHIP = 0
+local TOGGLE_DIST_SQR = TOGGLE_DIST ^ 2
+
+local matChipLister = CreateMaterial( MATERIAL_NAME, "UnlitGeneric", {
+    ["$basetexture"] = RENDER_TARGET_NAME,
+    ["$model"] = 1,
+} )
+
+local isValid = IsValid
+local rawset = rawset
+local rawget = rawget
+local _colorToHSV = ColorToHSV
+local _hsvToColor = HSVToColor
```

Comment:
you don't have to rename the variables, `local HSVToColor = HSVToColor` will work just fine. Lua picks the local over the global whenever it gets the chance. Renaming the functions only makes it harder for linting and later debugging imo.

Reply:
Stuff like string.format should be localized to something like `stringFormat` or `string_format` as they're still a table lookup ofc.

Reply:
It helps to use different names for in the off-chance that you do need to access the global again for some reason (unlikely, but always out there, and it's good to maintain consistency) and it makes it clearly obvious while reading the code that it is in fact a local version instead. Also, in this specific case I needed to also wrap the functions since they took unpacked number args instead of colors, hence the `_` prefix

Reply:
`string.format` is already localized

Reply:
string.format was an example, im aware it was already localized

Reply:
it's not wrapped, it's just used eg:
```lua
local function colorToHSV( color )
    return Color( _colorToHSV( color ) )
end
```
making your own function with the exact same name of a global is going to be an absolute pain for anyone that's going to maintain this code

Reply:
It's not the exact same name, that's kinda the whole point I was trying to make.

Reply:
considering `team.GetColor` gets turned into `teamGetColor` it's very feasable that `colorToHSV` is just a localized `ColorToHSV`

Reply:
Ah, right